File: Fundamentos da Programação Orientada a Objetos/Associacao_Agregacao/ex02/entidades/Pessoa.java
package entidades;

public class Pessoa {
	 private String nome;
	 private int idade;
	 private int identificacao;
	 private Endereco endereco;
	    
	    public Pessoa(String nome, int idade, int identificacao, Endereco endereco) {
	        this.nome = nome;
	        this.idade = idade;
	        this.identificacao = identificacao;
	        this.endereco = endereco;
	    }

	    public String getNome() {
	        return nome;
	    }

	    public void setNome(String nome) {
	        this.nome = nome;
	    }

	    public int getIdade() {
	        return idade;
	    }

	    public void setIdade(int idade) {
	        this.idade = idade;
	    }

	    public int getIdentificacao() {
	        return identificacao;
	    }

	    public void setIdentificacao(int identificacao) {
	        this.identificacao = identificacao;
	    }

	    public Endereco getEndereco() {
	        return endereco;
	    }

	    public void setEndereco(Endereco endereco) {
	        this.endereco = endereco;
	    }

	    @Override
	    public String toString() {
	        return "Pessoa{" +
	                "Nome: '" + nome + '\'' +
	                ", Idade: " + idade +
	                ", Identificacao: " + identificacao +
	                ", \nEndereco: " + endereco +
	                '}';
	    }
	}

File: Programação Web Front-End/Exercicio_Introdução_POO/exerc04/js/Pessoa.js
export default class Pessoa {
    #nome;
    #idade;
    #cpf;
    #endereco;

    constructor(nome= "", idade= 0, cpf= "", novoEndereco= "") {
        this.#nome = nome;
        this.#idade = idade;
        this.#cpf = cpf;
        this.#endereco = novoEndereco;
    }

    // Métodos getters e setters 
    get getNome() {
        return this.#nome;
    }
    set setNome(nome) {
        this.#nome = nome;
    }

    get getIdade() {
        return this.#idade;
    }
    set setIdade(idade) {
        this.#idade = idade;
    }

    get getCpf() {
        return this.#cpf;
    }
    set setCpf(cpf) {
        this.#cpf = cpf;
    }

    get getEndereco() {
        return this.#endereco;
    }
    set setEndereco(endereco) {
        this.#endereco = endereco;
    }

    // Método toString para exibir a pessoa
    toString() {
        return `Pessoa: ${this.#nome}, Idade: ${this.#idade}, CPF: ${this.#cpf}, \n ${this.#endereco.toString()}`;
    }
}